### ISSUE TITLE
Request for iOS Punctuation

### ISSUE THREAD

**Issue by jasonpearce** [cxam/passed.pw#1]
On a mobile phone keyboard, many punctuation characters are hidden two or three keyboards deep. It would be nice to have a third Punctuation option that included only the characters on the first two iOS keyboards, excluding characters from the third keyboard. Thank you for the consideration.

**Comment by jasonpearce** [comment]
These characters. http://www.globalnerdy.com/wordpress/wp-content/uploads/2014/12/ios-8-iphone-numbes-punctuation-keyboard.jpg

Perhaps you do the same for Android.

**Comment by cxam** [comment]
Thanks for the suggestion. This was something that I had in mind when I made the separation between **Common Punctuation** and **Other Characters**. Currently, the common punctuation option only includes a small set of characters (.,-:;?!) from the second level keyboard and the rest of the characters in the third level keyboard can be included by selecting the other characters option.

Since there are a few differences between Android and IOS for the second and third level keyboards, I could suggest expanding the common punctuation option to cover more characters by including only the common characters found in the second level keyboard of both operating systems to keep the options generic.

As it stands, this is possible but not _all_ the characters from the second level keyboard are included by the common punctuation option.

_Android keyboard levels_
![android_keyboard_levels](https://cloud.githubusercontent.com/assets/7763217/8512059/3ee96e98-232b-11e5-8ee4-b467b10cfdfa.png)

_iOS keyboard levels_
![ios_keyboard_levels](https://cloud.githubusercontent.com/assets/7763217/8512124/d9e17f48-232c-11e5-9c93-11ae57d8316f.png)
[Source (cropped from original)](http://figshare.com/articles/_Key_layout_for_Apple_iOS_5_1_s_touch_keyboard_for_the_iPhone_4S_Apple_Inc_used_in_for_the_simulated_typing_task_a_Letters_layout_b_numbers_layout_c_symbols_layout_/1167925)

**Comment by jasonpearce** [comment]
Thanks for considering this suggestion. 

Taking this one idea further, perhaps you create a "Mobile Friendly" checkbox. This option would 

a) create passwords using only the characters on the primary and second-level keyboards that are common to both iOS and Android

b) group/cluster the characters by keyboard level so that a user only has to switch keyboards once or twice.

Example for idea B:
While the Mobile Friendly password of "mk7fw6W((cx-w?5" is good, the user has to switch keyboards 11 times. Similar password entropy could be offered using the same characters by grouping them to each keyboard like so: "mkfwWcxw76((-?5". This latter example is much more mobile friendly and takes fewer strokes to enter since the user does not have to switch back and forth among keyboards.

Negatives of idea B:
I realize that grouping characters by keyboard makes the password a little less secure. I also assume this would be more difficult to code than my first suggestion (idea A). But it's something I think about each time I create a new password. I often use a tool like yours to generate a complex password and then manually group the generated content to make it more mobile friendly. It would be awesome if your tool could do it all for me.

Thank you:
I appreciate your simple solution and easy to remember URL. So thank you. Please just consider these ideas to be friendly feature requests from an early fan.

**Comment by cxam** [comment]
Thanks for clarifying and I'm glad you like the solution. 

I like the idea of mobile keyboard friendly passwords and as you've already mentioned, it does compromise security for convenience (albeit not extremely). I've made a note regarding this in the about page just so people are aware.

The common punctuation character set has been extended to included characters common in both second level keyboards on Android and iOS (a). I've also implemented the grouping suggestion (b) and have pushed it live. Let me know your thoughts, thanks.
